### ISSUE TITLE
Exception "type 'Null' is not a subtype of type 'num' in type cast" after call Purchases.getOfferings()

### ISSUE THREAD

**Issue by signmotion** [RevenueCat/purchases-flutter#295]
- [X] I have updated Purchases SDK to the latest version
- [X] I have read the [Contribution Guidelines](https://github.com/RevenueCat/purchases-flutter/blob/main/CONTRIBUTING.md)
- [X] I have searched the [Community](https://community.revenuecat.com)
- [X] I have read [docs.revenuecat.com](https://docs.revenuecat.com/)
- [X] I have searched for [existing Github issues](https://github.com/RevenueCat/purchases-flutter/issues)

**Describe the bug**
After call `await Purchases.getOfferings()` we get an exception
`type 'Null' is not a subtype of type 'num' in type cast`

1. Environment

   1. Output of `flutter doctor`
```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 2.8.1, on Microsoft Windows [Version 10.0.22000.434], locale ru-UA)
[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[√] Chrome - develop for the web
[√] Android Studio (version 2020.3)
[√] Connected device (3 available)
• No issues found!
```

   2. How widespread is the issue. Percentage of devices affected.
100%, real and virtual Android devices.

2. [Debug logs](https://docs.revenuecat.com/docs/debugging) that reproduce the issue
```
D/[Purchases] - DEBUG(21462): ℹ️ Debug logging enabled
D/[Purchases] - DEBUG(21462): ℹ️ SDK Version - 4.6.0
D/[Purchases] - DEBUG(21462): 👤 Initial App User ID - null
D/[Purchases] - DEBUG(21462): 👤 Identifying App User ID: $RCAnonymousID:41e419f3c3a8440dbf9a4b11b83bb4e9
D/[Purchases] - DEBUG(21462): ℹ️ Deleting old synced subscriber attributes that don't belong to $RCAnonymousID:41e419f3c3a8440dbf9a4b11b83bb4e9
D/[Purchases] - DEBUG(21462): ℹ️ App foregrounded
D/[Purchases] - DEBUG(21462): ℹ️ PurchaserInfo cache is stale, updating from network in foreground.
D/[Purchases] - DEBUG(21462): ℹ️ Offerings cache is stale, updating from network in foreground
D/[Purchases] - DEBUG(21462): 😻 Offerings updated from network.
D/[Purchases] - DEBUG(21462): ℹ️ Skipping updating pending purchase queue since BillingClient is not connected yet.
D/[Purchases] - DEBUG(21462): ℹ️ No subscriber attributes to synchronize.
D/[Purchases] - DEBUG(21462): ℹ️ Listener set
D/[Purchases] - DEBUG(21462): ℹ️ Sending latest PurchaserInfo to listener.

D/[Purchases] - DEBUG(21462): ℹ️ Starting connection for com.android.billingclient.api.BillingClientImpl@e0192dd

D/[Purchases] - DEBUG(21462): ℹ️ Billing Service Setup finished for com.android.billingclient.api.BillingClientImpl@e0192dd
D/[Purchases] - DEBUG(21462): ℹ️ Updating pending purchase queue
D/[Purchases] - DEBUG(21462): ℹ️ API request started: GET /subscribers/%24RCAnonymousID%3A41e419f3c3a8440dbf9a4b11b83bb4e9

D/[Purchases] - DEBUG(21462): ℹ️ API request completed with status: GET /subscribers/%24RCAnonymousID%3A41e419f3c3a8440dbf9a4b11b83bb4e9 304
D/[Purchases] - DEBUG(21462): ℹ️ PurchaserInfo updated, sending to listener.

D/[Purchases] - DEBUG(21462): ℹ️ API request started: GET /subscribers/%24RCAnonymousID%3A41e419f3c3a8440dbf9a4b11b83bb4e9/offerings
D/[Purchases] - DEBUG(21462): ℹ️ API request completed with status: GET /subscribers/%24RCAnonymousID%3A41e419f3c3a8440dbf9a4b11b83bb4e9/offerings 304
D/[Purchases] - DEBUG(21462): ℹ️ Requesting products from the store with identifiers: hp.99.1m, hp.1099.1y
D/[Purchases] - DEBUG(21462): ℹ️ Querying purchases
D/[Purchases] - DEBUG(21462): ℹ️ Cleaning previously sent tokens
D/[Purchases] - DEBUG(21462): ℹ️ Tokens already posted: []
D/[Purchases] - DEBUG(21462): ℹ️ Saving tokens []
D/[Purchases] - DEBUG(21462): ℹ️ Tokens already posted: []
D/[Purchases] - DEBUG(21462): ℹ️ Products request finished for hp.99.1m, hp.1099.1y
D/[Purchases] - DEBUG(21462): 💰 Retrieved skuDetailsList: SkuDetails: {"productId":"hp.1099.1y","type":"subs","title":"Годовая подписка (Счастье)","name":"Годовая подписка","price":"299,99 грн.","price_amount_micros":299990000,"price_currency_code":"UAH","description":"Все существующие и будущие варианты картин на год. Анимация и звук по тапу.","subscriptionPeriod":"P1Y","skuDetailsToken":"AEuhp4KneCpN9CFP4IGHzA6d3zyfehMuFCWB1zrxmDZwCpjwLLE57vYU7QgMYvU3YXXf"}, SkuDetails: {"productId":"hp.99.1m","type":"subs","title":"Ежемесячная подписка (Счастье)","name":"Ежемесячная подписка","price":"29,99 грн.","price_amount_micros":29990000,"price_currency_code":"UAH","description":"Все существующие и будущие варианты картин на месяц. Анимация и звук по \nтапу.","subscriptionPeriod":"P1M","skuDetailsToken":"AEuhp4LU9whZq6aEykT7aWoESkFhHGit3-3F1AqIJ8nIdGcNUGS-2qNXqBjb54e-kVfP"}
D/[Purchases] - DEBUG(21462): 💰 hp.1099.1y - SkuDetails: {"productId":"hp.1099.1y","type":"subs","title":"Годовая подписка (Счастье)","name":"Годовая подписка","price":"299,99 грн.","price_amount_micros":299990000,"price_currency_code":"UAH","description":"Все существующие и будущие варианты картин на год. Анимация и звук по тапу.","subscriptionPeriod":"P1Y","skuDetailsToken":"AEuhp4KneCpN9CFP4IGHzA6d3zyfehMuFCWB1zrxmDZwCpjwLLE57vYU7QgMYvU3YXXf"}
D/[Purchases] - DEBUG(21462): 💰 hp.99.1m - SkuDetails: {"productId":"hp.99.1m","type":"subs","title":"Ежемесячная подписка (Счастье)","name":"Ежемесячная подписка","price":"29,99 грн.","price_amount_micros":29990000,"price_currency_code":"UAH","description":"Все существующие и будущие варианты картин на месяц. Анимация и звук по \nтапу.","subscriptionPeriod":"P1M","skuDetailsToken":"AEuhp4LU9whZq6aEykT7aWoESkFhHGit3-3F1AqIJ8nIdGcNUGS-2qNXqBjb54e-kVfP"}

D/[Purchases] - DEBUG(21462): ℹ️ Vending PurchaserInfo from cache.
D/[Purchases] - DEBUG(21462): ℹ️ Checking if cache is stale AppInBackground false

D/[Purchases] - DEBUG(21462): ℹ️ Vending Offerings from cache
D/[Purchases] - DEBUG(21462): ℹ️ Checking if cache is stale AppInBackground false
```

3. Steps to reproduce, with a description of expected vs. actual behavior
```
    late Offerings offerings;
    try {
      offerings = await Purchases.getOfferings();
    } catch (ex) {
      print(ex);
    }
```

4. **Other information** (e.g. stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, etc.)

This seems to be happening on deserializing JSON like this in this place:
![image](https://user-images.githubusercontent.com/311074/149614316-01ea698b-5b6c-4934-9f33-b3c3fdc83678.png)

**Additional context**

Project in the RevenueCat: https://app.revenuecat.com/projects/7353a3c5

Subscription options on the Google Play Console:
![image](https://user-images.githubusercontent.com/311074/149614138-8dec79b6-8b67-4ec1-b481-42edfd4f8d2a.png)

**Comment by NatsuOnFire** [comment]
Fixed in #294

**Comment by aboedo** [comment]
Update here: release 3.9.1 has been released with the fix! 🎉  Thanks so much for reporting!
